ctapipe/reco/impact_distance.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+
+from typing import Union
+
+import numpy as np
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+
+from ..containers import ReconstructedGeometryContainer, SimulatedShowerContainer
+from ..coordinates import GroundFrame, TiltedGroundFrame, altaz_to_righthanded_cartesian
+from ..instrument.subarray import SubarrayDescription
+
+__all__ = ["shower_impact_distance", "impact_distance"]
+
+
+def impact_distance(point: np.ndarray, direction: np.ndarray, test_points: np.ndarray):
+    """Compute impact distance from a line defined by a point and a direction vector
+    with an array of points
+
+    Parameters
+    ----------
+    point: np.ndarray
+       point defining the line
+    direction: np.ndarray
+       direction vector
+    test_points: np.ndarray
+       array of test points
+    """
+    return np.linalg.norm(
+        np.cross(test_points - point, direction), axis=1
+    ) / np.linalg.norm(direction)
+
+
+def shower_impact_distance(
+    shower_geom: Union[ReconstructedGeometryContainer, SimulatedShowerContainer],
+    subarray: SubarrayDescription,
+):
+    """computes the impact distance of the shower axis to the telescope positions
+
+    Parameters
+    ----------
+    shower_geom: ReconstructedGeometryContainer
+        reconstructed shower geometry, must contain a core and alt/az position
+    subarray: SubarrayDescription
+        SubarrayDescription from which to extract the telescope positions
+
+    Returns
+    -------
+    np.ndarray:
+       array of impact distances to each telescope (packed by tel index)
+    """
+
+    # core position in the ground frame
+    core_position = np.array(
+        [
+            shower_geom.core_x.to_value(u.m),
+            shower_geom.core_y.to_value(u.m),
+            0,
+        ]
+    )
+
+    # sky direction of the shower (defines the shower axis)
+    sky_direction = altaz_to_righthanded_cartesian(
+        alt=shower_geom.alt, az=shower_geom.az
+    )
+
+    # telescope positions in the ground frame
+    telescope_positions = subarray.tel_coords.cartesian.xyz.to_value("m").T
+
+    return (
+        impact_distance(
+            point=core_position,
+            direction=sky_direction,
+            test_points=telescope_positions,
+        )
+        * u.m
+    )
+
+
+def shower_impact_distance_with_frames(
```

Comment:
Is this still needed now that we established that both are the same?

Reply:
probably not, but it is useful for the test.